friends/src/reducers/friendsReducer.js

```diff
@@ -0,0 +1,107 @@
+import {
+  LOADING,
+  LOGIN_SUCCESS,
+  LOGIN_FAILURE,
+  IS_FETCHING_FRIENDS,
+  FETCHING_FRIENDS_SUCCESS,
+  FETCHING_FRIENDS_FAILURE,
+  ADDING_FRIEND,
+  ADDING_FRIEND_SUCCESS,
+  ADDING_FRIEND_FAILURE,
+  DELETING_FRIEND,
+  DELETING_FRIEND_SUCCESS,
+  DELETING_FRIEND_FAILURE
+} from "../actions/actionTypes";
+
+const initialState = {
+  fetchingFriends: false,
+  savingFriend: false,
+  deletingFriend: false,
+  friends: [],
+  loggingIn: false,
+  error: ''
+};
+
+export const loginReducer = (state = initialState, action) => {
+  switch (action.type) {
```

Comment:
I don't really like how you use the same state for different reducers. Every reducer should just care about the slice of state it's managing. In this way it becomes kind of a mess to understand who is modifying what.